src/core/counter.h

```diff
@@ -0,0 +1 @@
+struct counter { uint64_t c; };
```

Comment:
I have no personal preference, I wasn't sure about the scope of `cdef`'s so this was my cautious approach to just mimic what I knew worked. I could imagine a C program wanting to read snabb counters so that would be an argument for having the header file. Since this also applies to the other header files in core as well I say lets move this into a seperate general discussion on whether we want C header files and in wjich cases.